aws-xray-recorder-sdk-core/src/main/java/com/amazonaws/xray/plugins/EC2MetadataFetcher.java

```diff
@@ -46,7 +47,7 @@
         AMI_ID,
     }
 
-    private static final int TIMEOUT_MILLIS = 2000;
+    private static final int TIMEOUT_MILLIS = 100;
```

Comment:
What do you think about keeping a higher (maybe not this high) read timeout and only lowering connect?

Reply:
I'm wondering wether there's a better detection mechanism for EC2, before we actually start up the plug-in... similar to how we detect Lambda environments from env vars.

Reply:
I have a feeling it's possible to speed up the positive case but we may be stuck on the negative case. EC2 provides low level customization of everything so env cars, file system can't be relied. I found a post where the hostname will end with an Amazon suffix by default but that's only if it's not overridden...

@willarmiros Out of curiosity what JVM did you test the timeout on? Java 11+ have a completely revamped network stack and I wonder if this may have improved. I think most tools fail fast when an IP is completely unreachable, which should be what's happening with metadata service which uses a link local, not public, IP address.

Reply:
@awssandra as Rag said, can't rely on much other than IMDS. There's no single file to check for that exists on all instances on Windows & Linux. Hostname is overridden in VPCs which many EC2 Customers use.

@anuraaga I tested the timeout using Java 8. I can change it to have say a 100 ms Connect timeout and 1 second read timeout?

Reply:
> I think most tools fail fast when an IP is completely unreachable

Do you mean they throw an exception before even timing out? Would this reliably happen in Java 8 & 11?

Reply:
Ah I just meant that newer network stack might through an exception right away regardless of the timeout value. Not a big deal mostly just curious :)